### ISSUE TITLE
Configuration File Switching Failure

### ISSUE THREAD

**Issue by barryytm** [fgpv-vpgf/fgpv-vpgf#2442]
`index-samples` fails to load another config file.  The following error message appears on the console when trying to select a different config.

![capture](https://user-images.githubusercontent.com/19259169/31946591-4375eeac-b8a0-11e7-8e1c-da861815ec66.PNG)

**Comment by milesap** [comment]
Likely nothing to do with extension loader. [config.service.js](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/develop/src/app/core/config.service.js) was recently overhauled. 

I'm thinking `config.getSync` should actually be removed from our code since we can never really be sure the config is loaded.

**Comment by AleksueiR** [comment]
Loading `index-fgp-en.html?` fails in IE Edge as well with

>geoService failed to assemble the map with error Error: Attempted to access config synchronously before loading completed. Either use the promise based API or wait for rvReady.

>Microsoft Edge 38.14393.1066.0
>Microsoft EdgeHTML 14.14393

**Comment by barryytm** [comment]
@milesap is right about our code making assumptions on config being already loaded in a lot of places.  I have been trying to fix it by replacing `config.getSync` with `configService.getAsync`.  However this will not be a quick fix due to the time needed to analyse which dependencies are affected the code calling `config.getSync`.  Anyone got a suggestion on how this could be fixed in a faster way or should I just keep going with this approach?

**Comment by milesap** [comment]
Just blowing some hot air out of my tail pipe here but you could try an approach where when a config load/reload is triggered you halt that operation, tell the config service to get the config and when it comes back then execute the actual config load/reload.

Eliminating `config.getSync` would be a monumental task - best left for `v4.0.0` (but by then everything will be Angular 10 and Observables anyway).

**Comment by alyec** [comment]
> when a config load/reload is triggered you halt that operation

I'd just suggest rewording that from halt which threw me off a bit, to something like defer (although we don't like using that term around promises).  From our discussion I like the idea and didn't even realize our code was not doing this.  I think it makes a lot of sense to not swap out the old config until the new one is ready (i.e. downloaded and parsed).
